modules/tracktion_engine/timestretch/tracktion_RubberBand.cpp

```diff
@@ -0,0 +1,365 @@
+ #pragma GCC diagnostic ignored "-Wpedantic"
+ #pragma GCC diagnostic ignored "-Wunknown-pragmas"
+#endif
+
+#define Point CarbonDummyPointName
+#define Component CarbonDummyCompName
+
+//#include "rubberband/rubberband-c.cpp"
+#include "../3rd_party/rubberband/src/StretcherProcess.cpp"
+#include "../3rd_party/rubberband/src/StretchCalculator.cpp"
+#include "../3rd_party/rubberband/src/StretcherChannelData.cpp"
+#include "../3rd_party/rubberband/src/StretcherImpl.cpp"
+#include "../3rd_party/rubberband/src/system/Allocators.cpp"
+#include "../3rd_party/rubberband/src/system/sysutils.cpp"
+#include "../3rd_party/rubberband/src/system/Thread.cpp"
+#include "../3rd_party/rubberband/src/system/VectorOpsComplex.cpp"
+// #include "rubberband/pommier/neon_mathfun.h"
+// #include "rubberband/pommier/sse_mathfun.h"
+#include "../3rd_party/rubberband/src/dsp/AudioCurveCalculator.cpp"
+#include "../3rd_party/rubberband/src/dsp/FFT.cpp"
+#include "../3rd_party/rubberband/src/dsp/Resampler.cpp"
+#include "../3rd_party/rubberband/src/base/Profiler.cpp"
+#include "../3rd_party/rubberband/src/audiocurves/CompoundAudioCurve.cpp"
+#include "../3rd_party/rubberband/src/audiocurves/ConstantAudioCurve.cpp"
+#include "../3rd_party/rubberband/src/audiocurves/HighFrequencyAudioCurve.cpp"
+#include "../3rd_party/rubberband/src/audiocurves/PercussiveAudioCurve.cpp"
+#include "../3rd_party/rubberband/src/audiocurves/SilentAudioCurve.cpp"
+#include "../3rd_party/rubberband/src/audiocurves/SpectralDifferenceAudioCurve.cpp"
+#include "../3rd_party/rubberband/src/speex/resample.c"
+// #include "rubberband/jni/Rubbersrc/BandStretcherJNI.cpp"
+
+#include "../3rd_party/rubberband/src/RubberBandStretcher.cpp"
+
+#undef Point
+#undef Component
+
+#ifdef __GNUC__
+ #pragma GCC diagnostic pop
+#endif
+
+bool RubberStretcher::setSpeedAndPitch(float speedRatio, float semitonesUp)
+{
+    float pitch = juce::jlimit(0.25f, 4.0f, tracktion_engine::Pitch::semitonesToRatio (semitonesUp));
+
+    // Not sure what to do with sync
+
+    // bool sync = (elastiqueMode == TimeStretcher::elastiquePro) ? elastiqueProOptions.syncTimeStrPitchShft : false;
+
+    // TODO: Assert this works
+    thisRubberBandStretcher.setPitchScale(pitch);
+    thisRubberBandStretcher.setTimeRatio(speedRatio);
+
+    int r = (thisRubberBandStretcher.getPitchScale() == pitch && thisRubberBandStretcher.getTimeRatio() == speedRatio);
+    jassert(r == 0); juce::ignoreUnused(r);
+
+    return r == 0;
+}
+
+int RubberStretcher::getFramesNeeded() const
+{
+    const int framesNeeded = thisRubberBandStretcher.available();
+    jassert(framesNeeded <= maxFramesNeeded);
+    return framesNeeded;
+}
+
+int RubberStretcher::getMaxFramesNeeded() const
+{
+    return maxFramesNeeded;
+}
+
+void RubberStretcher::processData(const float* const* inChannels, int numSamples, float* const* outChannels)
+{
+    thisRubberBandStretcher.process((float**)outChannels, numSamples, false);
```

Comment:
This doesn't look right to me. The docs say `Provide a block of "samples" sample frames for processing`, outChannels is where the output data should be, i.e. the time-stretched/pitched data.